### PR TITLE
Graphclust update

### PR DESCRIPTION
**Pull request by mmiladi**

### CODE REVIEW

File: tools/GraphClust/Preprocessing/splitSHAPE.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
     lines = content.split('\n')
     for line in lines:
         if pattern.match(line):
-            line = line.replace('>','').strip()
+            line = line.replace('>','').split()[0]
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/splitSHAPE.py]: no stripping here?

File: tools/GraphClust/CollectResults/evaluation.py [hunk 2]
```diff
@@ -10,56 +10,65 @@ def sh(script):
...
 toWrite = ""
 for i in range(len(listOfClusters)):
-    toWrite += listOfClasses[i] + "\t" + listOfClusters[i] + '\n'
+    toWrite += listOfHeaders[i] + "\t" + listOfClusters[i] + '\n'
 with open("RESULTS/fullTab.tabular", "w") as full:
     full.write(toWrite)
 
 
 pattern = re.compile("^RF.*$")
+print (listOfHeaders[0])
```
Comment (bgruening) [inline tools/GraphClust/CollectResults/evaluation.py]: This should be removed I guess?

File: tools/GraphClust/CollectResults/evaluation.py [hunk 3]
```diff
@@ -10,56 +10,65 @@ def sh(script):
...
 with open(dataNames, "r") as names:
     for line in names.readlines():
         fullUniqeId = line.split()[3]
-        rnaClass, sep, tail = fullUniqeId.partition("_")
-        rnaClass, sep, tail = fullUniqeId.partition("_")
-        if fullUniqeId not in blackList:
-            listOfClasses.append(rnaClass)
+        fullHeader = ''
+        if len(line.split()) >= 6:
+            fullHeader = line.split()[5]
```
Comment (bgruening) [inline tools/GraphClust/CollectResults/evaluation.py]: if you move the split of of the condition and save it, you don't need to do it twice

File: tools/GraphClust/CollectResults/evaluation.py [hunk 4]
```diff
@@ -10,56 +10,65 @@ def sh(script):
...
 
 listOfClusters = []
-listOfClasses = []
+listOfHeaders = []
+headersNames = set()
 cluster_seqs_stats_path = "RESULTS/*.cluster.all"
 cluster_seqs_stats_files = glob.glob(cluster_seqs_stats_path)
 
+with open(dataNames, "r") as names:
+    for line2 in names.readlines():
```
Comment (bgruening) [inline tools/GraphClust/CollectResults/evaluation.py]: for line2 in names:

File: tools/GraphClust/Preprocessing/splitSHAPE.py (hunk 1 above)
Comment (mmiladi) [inline tools/GraphClust/Preprocessing/splitSHAPE.py]: Before it was assigning the full line now only the first element. I guess first element of the split is independent of striping, right?

File: tools/GraphClust/CollectResults/evaluation.py (hunk 4 above)
Comment (mmiladi) [inline tools/GraphClust/CollectResults/evaluation.py]: Agree, will fix it. But don't blame me for the coding style here :laughing:

File: tools/GraphClust/Preprocessing/splitSHAPE.py (hunk 1 above)
Comment (bgruening) [inline tools/GraphClust/Preprocessing/splitSHAPE.py]: Yes, but if you expect whitespaces in the first token, strip() it :)

**Comment by bgruening** [comment]
This will not help, Travis has problems, they are aware of it and working on it :)

**Comment by bgruening** [comment]
I hope you have used the `--update_test_data` option. This makes everything way easier.
Thanks for the update!

**Pull request merged**